static/app/utils/orgRole.tsx

```diff
@@ -0,0 +1,12 @@
+import {OrgRole} from 'sentry/types';
+
+export function getTopOrgRole(memberOrgRoles: string[], orgRoleList: OrgRole[]) {
+  // sort by ascending index
+  const orgRoleMap = orgRoleList.reduce((acc, role, index) => {
+    acc[role.id] = {index, role};
+    return acc;
+  }, {});
+  memberOrgRoles.sort((a, b) => orgRoleMap[b].index - orgRoleMap[a].index);
+
+  return orgRoleMap[memberOrgRoles[0]] ? orgRoleMap[memberOrgRoles[0]].role : undefined;
```

Comment:
[Optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) is nifty way to simplify this.

```suggestion
  return orgRoleMap[memberOrgRoles[0]]?.roles;
```